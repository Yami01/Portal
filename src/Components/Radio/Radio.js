/**
 *
 * Radio Button
 * @format
 * @flow
 *
 */

import type { AbstractComponent, Node } from "react"
import React, { memo } from "react"
import { RadioButton as RNRadioButton } from "react-native-paper"
import type { PropsType } from "./types"
import Colors from "react-native/Libraries/NewAppScreen/components/Colors"

const Radio: AbstractComponent<PropsType> = memo((props: PropsType): Node => {
  const { checked, onPress, value, color, uncheckedColor, disabled, testID } =
    props

  return (
    <RNRadioButton.Android
      {...testID}
      accessibilityRole="radio"
      status={checked ? 'checked' : 'unchecked'}
      value={value}
      onPress={onPress}
      color={color || Colors.primary}
      uncheckedColor={uncheckedColor || Colors.primary}
      disabled={disabled}
    />
  )
})

export default Radio
