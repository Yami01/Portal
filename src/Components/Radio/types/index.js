/**
 *
 * @format
 * @flow
 *
 */

export type PropsType = {
  accessibilityLabel: string,
  accessibilityHint: string,
  value: string,
  checked?: boolean,
  onPress: (boolean) => void,
  color?: string,
  uncheckedColor?: string,
  disabled?: boolean,
  testID?: string,
};
