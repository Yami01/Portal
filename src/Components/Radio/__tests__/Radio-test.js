/**
 *
 * Radio Test
 * @format
 * @flow
 *
 */

import React from 'react'
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react-native'

import Radio from '../Radio'

describe('Radio', () => {
  const testId = 'Radio Test Id'
  const onPressMock = jest.fn()
  const value = 'Test Value 1'

  test('Radio Render', () => {
    const tree = renderer
      .create(<Radio onPress={onPressMock} value={value} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Radio Disable Render', () => {
    const tree = renderer
      .create(<Radio onPress={onPressMock} value={value} disabled />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Radio Press', () => {
    const { queryByTestId } = render(
      <Radio testId={testId} onPress={onPressMock} value={value} />,
    )

    const button = queryByTestId(testId)
    fireEvent.press(button)
    expect(onPressMock).toHaveBeenCalled()
  })
})
