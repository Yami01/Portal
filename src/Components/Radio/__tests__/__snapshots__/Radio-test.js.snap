// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Radio Disable Render 1`] = `
<View
  accessibilityLiveRegion="polite"
  accessibilityRole="radio"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": true,
    }
  }
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 18,
      },
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "borderColor": "rgba(0, 0, 0, 0.26)",
        "borderRadius": 10,
        "borderWidth": 2,
        "height": 20,
        "margin": 8,
        "width": 20,
      }
    }
  />
</View>
`;

exports[`Radio Radio Render 1`] = `
<View
  accessibilityLiveRegion="polite"
  accessibilityRole="radio"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": false,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 18,
      },
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "borderColor": "#E14032",
        "borderRadius": 10,
        "borderWidth": 2,
        "height": 20,
        "margin": 8,
        "width": 20,
      }
    }
  />
</View>
`;
